### DISCUSSION TITLE
Is it possible to somehow implement BoS to control when to press it?

### DISCUSSION THREAD

**Discussion by Dragonfo** [Hekili/hekili#1464] [General]
Hello. After playing on FDK, I liked everything, except that the rotation always suggests using BoS. There are moments in the same raid where you do not want to immediately press BoS from the beginning of the pool, but would like to press after 30-40 seconds of the battle. I tried adding more rotation squares up to 5 pieces, but there is such a long rotation chain in terms of BoS activation that it seems to me that all 10 squares will not be enough. So I had a question, is there a way for us to conditionally reach 50+rp, the addon offers BoS but let's say in the defense square, so that such abilities as (Pillar of frost, Breath of sindragosa, Empower Rune Weapon) are located there. In simple words, how to try to control BoS. If you look at it this way, then the rotation consists of 3 buttons and 4 optional buttons - this is Remorseless Winter. Maybe someone knows how to solve this problem? or can someone tell me a way how to remove these 3 abilities that I listed above from the rotation, but without breaking the rotation. Thanks in advance.

**Comment by Hekili** [comment]
> There are moments in the same raid where you do not want to immediately press BoS from the beginning of the pool, but would like to press after 30-40 seconds of the battle.

This is what Toggles are for; you turn off Cooldowns (including Breath of Sindragosa) until you want to use them.

> So I had a question, is there a way for us to conditionally reach 50+rp,

/hekili > Frost > Runic Power for Breath of Sindragosa ?

> the addon offers BoS but let's say in the defense square, so that such abilities as (Pillar of frost, Breath of sindragosa, Empower Rune Weapon) are located there.

Are there ways to make abilities that are not defensives to show up in the defensives display?  Yes.  But I don't think it's what you actually want and I don't know why you'd want it.

> If you look at it this way, then the rotation consists of 3 buttons and 4 optional buttons - this is Remorseless Winter. Maybe someone knows how to solve this problem?

You can try using /hekili > Toggles > Cooldowns > Shown Separately and they'll appear in a different display, but that is not necessarily compatible with all specs.

> or can someone tell me a way how to remove these 3 abilities that I listed above from the rotation, but without breaking the rotation.

You can disable abilities you don't want the addon to recommend and then you can test to see if the output is useful to you.  /hekili > Abilities > [pick the ability] > Disabled (checked).

I recommend familiarizing yourself with /hekili > Toggles.

**Reply by Dragonfo** [reply]
Thanks for the good tip with Cooldowns. This feature helped me a lot. But there are two problems. The 1st problem is that when I moved abilities + covenant ability to Cooldowns, my rotation broke a little at the beginning of its work. This is manifested by the fact that the addon stops trying to immediately make 50+ runic power, and immediately prompts me to use Frost Strike when I don’t even have 30+ runic power yet. If it would be possible to somehow beat this so that the addon would further strive to collect runic power under BoS, then it would be great. 2 The problem is that if I turn on the Cooldown: Show Separately - Use Actual Cooldowns checkbox, then the rotation icons will no longer be displayed in Primary and will only be shown to me in Cooldown. If I click buttons which are shown in Cooldown then they will be shown in Primary. I would be grateful if you direct me how to solve these small problems, and this will be enough for me to be comfortable playing).

**Reply by Hekili** [reply]
> This is manifested by the fact that the addon stops trying to immediately make 50+ runic power, and immediately prompts me to use Frost Strike when I don’t even have 30+ runic power yet.

This is intentional.  If you're not going to use a cooldown, the resource pooling associated with that cooldown generally becomes inactive.  Otherwise, you can do really weird things to prepare for BoS (or any other cooldown) while not actually planning to use it any time soon.

>  If it would be possible to somehow beat this so that the addon would further strive to collect runic power under BoS, then it would be great.

Why do you want to pool at a time you're not going to use BoS?  

Ultimately, if you want a different priority, you'd have to write a different priority -- and I don't offer support for that because (1) there's a lot of nuance and (2) it would take way too much of my time.

> The problem is that if I turn on the Cooldown: Show Separately - Use Actual Cooldowns checkbox, then the rotation icons will no longer be displayed in Primary and will only be shown to me in Cooldown.

Don't do that, then.  (Note:  I said nothing about the Use Actual Cooldowns feature in that section, and I don't recommend messing with it.)  I pointed you to the Toggles section, here:

![image](https://user-images.githubusercontent.com/9438134/161396019-c011b3d9-20b9-4f79-aea3-b0eef0da9824.png)

**Comment by Dragonfo** [comment]
I'll try to explain briefly. BoS is a very strong cooldown. We can't just push it from the pool. There are moments on the bosses where there will be a lot of targets in 20-30 seconds, where it is more profitable for us to press BoS and get many times more dps. If the bosses were all linear where he stands and does nothing, then it makes sense to push BoS from the pool. But the new raid is modeled so we want to use BoS later. Since BL can be pressed later, and the BoS efficiency will again be worse. I have attached a screenshot showing that BoS is used very late on bosses. And this is not one exception and not one boss in the new raid. Where the gaps can be shorter and can be longer.That's why I'm looking for a way out of the situation so I can keep my BoS. 
In the old raids, there were very small gaps between such fights, and it didn’t matter when BoS was pressed, but now the bosses have become more dynamic, and I want to leave my bursts for later moments. So far, I see a solution to either transfer to Cooldowns or increase the rotation display to a maximum value of 10 squares. I don't see any other way yet to delay my bursts
![1](https://user-images.githubusercontent.com/63850853/161396969-13c57081-24d7-441f-a00b-000eb6c5666a.jpg)

**Comment by Hekili** [comment]
Yes, I understand what BoS is and I understand that there are fights where you don't use all your cooldowns at the pull.

That's why you can toggle Cooldowns off, then press the toggle button (ALT-SHIFT-R, by default) to turn them back on when you want the addon to recommend them again.
